File: backend/models/nutrition.go
package models

import (
	"time"

	"gorm.io/gorm"
)

// ==========================================
// NUTRITION / GIZI MANAGEMENT
// ==========================================

// Nutrition Menu Category Constants
const (
	NutritionCategoryMakananPokok = "makanan_pokok" // Nasi, bubur, dll
	NutritionCategoryLauk         = "lauk"          // Lauk pauk
	NutritionCategorySayur        = "sayur"         // Sayuran
	NutritionCategoryBuah         = "buah"          // Buah-buahan
	NutritionCategorySnack        = "snack"         // Makanan ringan
	NutritionCategoryMinuman      = "minuman"       // Minuman
	NutritionCategorySuplemen     = "suplemen"      // Suplemen gizi
	NutritionCategoryLainnya      = "lainnya"       // Lainnya
)

// Nutrition Diet Type Constants
const (
	NutritionDietBiasa         = "biasa"          // Diet biasa / umum
	NutritionDietLunak         = "lunak"          // Makanan lunak
	NutritionDietSaring        = "saring"         // Makanan saring
	NutritionDietCair          = "cair"           // Makanan cair
	NutritionDietDM            = "dm"             // Diet Diabetes Mellitus
	NutritionDietRendahGaram   = "rendah_garam"   // Diet Rendah Garam
	NutritionDietRendahLemak   = "rendah_lemak"   // Diet Rendah Lemak
	NutritionDietTinggiKalori  = "tinggi_kalori"  // Diet Tinggi Kalori Tinggi Protein
	NutritionDietRendahProtein = "rendah_protein" // Diet Rendah Protein
	NutritionDietRendahPurin   = "rendah_purin"   // Diet Rendah Purin
	NutritionDietLambung       = "lambung"        // Diet Lambung
	NutritionDietJantung       = "jantung"        // Diet Jantung
	NutritionDietGinjal        = "ginjal"         // Diet Ginjal
	NutritionDietHati          = "hati"           // Diet Hati
	NutritionDietLainnya       = "lainnya"        // Diet khusus lainnya
)

// Nutrition Meal Time Constants
const (
	NutritionMealPagi       = "pagi"        // Makan Pagi
	NutritionMealSnackPagi  = "snack_pagi"  // Snack Pagi
	NutritionMealSiang      = "siang"       // Makan Siang
	NutritionMealSnackSore  = "snack_sore"  // Snack Sore
	NutritionMealSore       = "sore"        // Makan Sore/Malam
	NutritionMealSnackMalam = "snack_malam" // Snack Malam
)

// NutritionMenu represents a food/drink item in the nutrition master data
// Master Menu Makanan
type NutritionMenu struct {
	ID        uint           `gorm:"primarykey" json:"id"`
	CreatedAt time.Time      `json:"created_at"`
	UpdatedAt time.Time      `json:"updated_at"`
	DeletedAt gorm.DeletedAt `gorm:"index" json:"-"`

	// Basic Information
	Code        string `gorm:"not null;uniqueIndex;size:20" json:"code"` // Kode menu (MNU-001)
	Name        string `gorm:"not null;size:200" json:"name"`            // Nama menu
	Description string `gorm:"type:text" json:"description,omitempty"`   // Deskripsi menu
	Category    string `gorm:"not null;size:50;index" json:"category"`   // makanan_pokok, lauk, sayur, buah, snack, minuman, suplemen
	DietTypes   string `gorm:"type:text" json:"diet_types,omitempty"`    // JSON array: ["biasa", "dm", "rendah_garam"] - diet yang cocok

	// Nutritional Information (per porsi)
	Calories     float64 `gorm:"type:decimal(10,2);default:0" json:"calories"`     // Kalori (kkal)
	Protein      float64 `gorm:"type:decimal(10,2);default:0" json:"protein"`      // Protein (gram)
	Fat          float64 `gorm:"type:decimal(10,2);default:0" json:"fat"`          // Lemak (gram)
	Carbohydrate float64 `gorm:"type:decimal(10,2);default:0" json:"carbohydrate"` // Karbohidrat (gram)
	Fiber        float64 `gorm:"type:decimal(10,2);default:0" json:"fiber"`        // Serat (gram)
	Sodium       float64 `gorm:"type:decimal(10,2);default:0" json:"sodium"`       // Natrium (mg)

	// Serving
	ServingSize string  `gorm:"size:50" json:"serving_size,omitempty"`          // Ukuran porsi (contoh: "1 porsi", "200ml")
	UnitPrice   float64 `gorm:"type:decimal(15,2);default:0" json:"unit_price"` // Harga per porsi

	// Status
	IsActive bool   `gorm:"default:true" json:"is_active"`
	Notes    string `gorm:"type:text" json:"notes,omitempty"`

	// Relations
	Ingredients []NutritionMenuIngredient `gorm:"foreignKey:MenuID;constraint:OnUpdate:CASCADE,OnDelete:CASCADE" json:"ingredients,omitempty"`
}

func (NutritionMenu) TableName() string {
	return "nutrition_menus"
}

// NutritionPackage represents a meal package (collection of menu items)
// Master Paket Makanan
type NutritionPackage struct {
	ID        uint           `gorm:"primarykey" json:"id"`
	CreatedAt time.Time      `json:"created_at"`
	UpdatedAt time.Time      `json:"updated_at"`
	DeletedAt gorm.DeletedAt `gorm:"index" json:"-"`

	// Basic Information
	Code        string `gorm:"not null;uniqueIndex;size:20" json:"code"` // Kode paket (PKT-001)
	Name        string `gorm:"not null;size:200" json:"name"`            // Nama paket (contoh: "Paket Diet DM Pagi")
	Description string `gorm:"type:text" json:"description,omitempty"`   // Deskripsi paket
	DietType    string `gorm:"not null;size:50;index" json:"diet_type"`  // Jenis diet paket ini
	MealTime    string `gorm:"not null;size:20;index" json:"meal_time"`  // Waktu makan: pagi, siang, sore, snack_pagi, snack_sore

	// Computed Nutritional Info (total dari semua item)
	TotalCalories     float64 `gorm:"type:decimal(10,2);default:0" json:"total_calories"`
	TotalProtein      float64 `gorm:"type:decimal(10,2);default:0" json:"total_protein"`
	TotalFat          float64 `gorm:"type:decimal(10,2);default:0" json:"total_fat"`
	TotalCarbohydrate float64 `gorm:"type:decimal(10,2);default:0" json:"total_carbohydrate"`

	// Pricing
	Price float64 `gorm:"type:decimal(15,2);default:0" json:"price"` // Harga paket (bisa override total item price)

	// Status
	IsActive bool   `gorm:"default:true" json:"is_active"`
	Notes    string `gorm:"type:text" json:"notes,omitempty"`

	// Relations
	Items []NutritionPackageItem `gorm:"foreignKey:PackageID" json:"items,omitempty"`
}

func (NutritionPackage) TableName() string {
	return "nutrition_packages"
}

// NutritionPackageItem represents a single menu item within a package
type NutritionPackageItem struct {
	ID        uint           `gorm:"primarykey" json:"id"`
	CreatedAt time.Time      `json:"created_at"`
	UpdatedAt time.Time      `json:"updated_at"`
	DeletedAt gorm.DeletedAt `gorm:"index" json:"-"`

	PackageID uint              `gorm:"not null;index:idx_package_menu" json:"package_id"`
	Package   *NutritionPackage `gorm:"foreignKey:PackageID;constraint:OnUpdate:CASCADE,OnDelete:CASCADE" json:"-"`

	MenuID uint           `gorm:"not null;index:idx_package_menu" json:"menu_id"`
	Menu   *NutritionMenu `gorm:"foreignKey:MenuID" json:"menu,omitempty"`

	Quantity float64 `gorm:"type:decimal(10,2);default:1" json:"quantity"` // Jumlah porsi
	Notes    string  `gorm:"type:text" json:"notes,omitempty"`             // Catatan khusus
}

func (NutritionPackageItem) TableName() string {
	return "nutrition_package_items"
}

// NutritionOrder represents a nutrition/meal order for an inpatient per day
// Order Gizi Pasien Rawat Inap
type NutritionOrder struct {
	ID        uint           `gorm:"primarykey" json:"id"`
	CreatedAt time.Time      `json:"created_at"`
	UpdatedAt time.Time      `json:"updated_at"`
	DeletedAt gorm.DeletedAt `gorm:"index" json:"-"`

	// Patient & Visit Reference
	VisitID   uint     `gorm:"not null;index" json:"visit_id"`
	Visit     *Visit   `gorm:"foreignKey:VisitID" json:"visit,omitempty"`
	PatientID uint     `gorm:"not null;index" json:"patient_id"`
	Patient   *Patient `gorm:"foreignKey:PatientID" json:"patient,omitempty"`

	// Order Details
	OrderDate time.Time `gorm:"not null;index" json:"order_date"`               // Tanggal menu berlaku
	MealTime  string    `gorm:"not null;size:20;index" json:"meal_time"`        // pagi, siang, sore, snack_pagi, snack_sore, snack_malam
	DietType  string    `gorm:"not null;size:50" json:"diet_type"`              // Jenis diet
	Status    string    `gorm:"not null;size:20;default:'draft'" json:"status"` // draft, confirmed, preparing, delivered, cancelled

	// Package reference (optional - if using package)
	PackageID *uint             `gorm:"index" json:"package_id,omitempty"`
	Package   *NutritionPackage `gorm:"foreignKey:PackageID" json:"package,omitempty"`

	// Room & Bed info (denormalized for kitchen display)
	RoomName string `gorm:"size:100" json:"room_name,omitempty"` // Nama ruangan
	BedName  string `gorm:"size:50" json:"bed_name,omitempty"`   // Nama bed

	// Ordered by (ahli gizi)
	OrderedByID *uint     `gorm:"index" json:"ordered_by_id,omitempty"`
	OrderedBy   *Employee `gorm:"foreignKey:OrderedByID" json:"ordered_by,omitempty"`

	// Delivery tracking
	PreparedAt  *time.Time `json:"prepared_at,omitempty"`
	DeliveredAt *time.Time `json:"delivered_at,omitempty"`
	DeliveredBy *uint      `gorm:"index" json:"delivered_by_id,omitempty"`

	// Notes
	AllergyNotes string `gorm:"type:text" json:"allergy_notes,omitempty"` // Catatan alergi makanan
	SpecialNotes string `gorm:"type:text" json:"special_notes,omitempty"` // Catatan khusus lainnya

	// Relations
	Items []NutritionOrderItem `gorm:"foreignKey:OrderID" json:"items,omitempty"`
}

func (NutritionOrder) TableName() string {
	return "nutrition_orders"
}

// NutritionOrderItem represents a single menu item in a nutrition order
type NutritionOrderItem struct {
	ID        uint           `gorm:"primarykey" json:"id"`
	CreatedAt time.Time      `json:"created_at"`
	UpdatedAt time.Time      `json:"updated_at"`
	DeletedAt gorm.DeletedAt `gorm:"index" json:"-"`

	OrderID uint            `gorm:"not null;index" json:"order_id"`
	Order   *NutritionOrder `gorm:"foreignKey:OrderID;constraint:OnUpdate:CASCADE,OnDelete:CASCADE" json:"-"`

	MenuID uint           `gorm:"not null;index" json:"menu_id"`
	Menu   *NutritionMenu `gorm:"foreignKey:MenuID" json:"menu,omitempty"`

	Quantity float64 `gorm:"type:decimal(10,2);default:1" json:"quantity"` // Jumlah porsi
	Notes    string  `gorm:"type:text" json:"notes,omitempty"`             // Catatan per item
}

func (NutritionOrderItem) TableName() string {
	return "nutrition_order_items"
}

// ==========================================
// HELPER LABELS
// ==========================================

var NutritionCategoryLabels = map[string]string{
	NutritionCategoryMakananPokok: "Makanan Pokok",
	NutritionCategoryLauk:         "Lauk Pauk",
	NutritionCategorySayur:        "Sayuran",
	NutritionCategoryBuah:         "Buah-buahan",
	NutritionCategorySnack:        "Makanan Ringan",
	NutritionCategoryMinuman:      "Minuman",
	NutritionCategorySuplemen:     "Suplemen Gizi",
	NutritionCategoryLainnya:      "Lainnya",
}

var NutritionDietTypeLabels = map[string]string{
	NutritionDietBiasa:         "Diet Biasa",
	NutritionDietLunak:         "Makanan Lunak",
	NutritionDietSaring:        "Makanan Saring",
	NutritionDietCair:          "Makanan Cair",
	NutritionDietDM:            "Diet Diabetes Mellitus",
	NutritionDietRendahGaram:   "Diet Rendah Garam",
	NutritionDietRendahLemak:   "Diet Rendah Lemak",
	NutritionDietTinggiKalori:  "Diet Tinggi Kalori Tinggi Protein",
	NutritionDietRendahProtein: "Diet Rendah Protein",
	NutritionDietRendahPurin:   "Diet Rendah Purin",
	NutritionDietLambung:       "Diet Lambung",
	NutritionDietJantung:       "Diet Jantung",
	NutritionDietGinjal:        "Diet Ginjal",
	NutritionDietHati:          "Diet Hati",
	NutritionDietLainnya:       "Diet Lainnya",
}

var NutritionMealTimeLabels = map[string]string{
	NutritionMealPagi:       "Makan Pagi",
	NutritionMealSnackPagi:  "Snack Pagi",
	NutritionMealSiang:      "Makan Siang",
	NutritionMealSnackSore:  "Snack Sore",
	NutritionMealSore:       "Makan Sore",
	NutritionMealSnackMalam: "Snack Malam",
}

// Nutrition Ingredient Unit Constants
const (
	NutritionIngredientUnitGram        = "gram"
	NutritionIngredientUnitMl          = "ml"
	NutritionIngredientUnitPcs         = "pcs"
	NutritionIngredientUnitBuah        = "buah"
	NutritionIngredientUnitSendokMakan = "sendok_makan"
	NutritionIngredientUnitSendokTeh   = "sendok_teh"
)

var NutritionIngredientUnitLabels = map[string]string{
	NutritionIngredientUnitGram:        "Gram",
	NutritionIngredientUnitMl:          "Mililiter",
	NutritionIngredientUnitPcs:         "Pcs",
	NutritionIngredientUnitBuah:        "Buah",
	NutritionIngredientUnitSendokMakan: "Sendok Makan",
	NutritionIngredientUnitSendokTeh:   "Sendok Teh",
}

// NutritionIngredient represents master ingredient data for nutrition menus.
type NutritionIngredient struct {
	ID        uint           `gorm:"primarykey" json:"id"`
	CreatedAt time.Time      `json:"created_at"`
	UpdatedAt time.Time      `json:"updated_at"`
	DeletedAt gorm.DeletedAt `gorm:"index" json:"-"`

	Code          string  `gorm:"not null;uniqueIndex;size:20" json:"code"`            // Kode bahan (BAH-001)
	Name          string  `gorm:"not null;size:200;index" json:"name"`                 // Nama bahan
	Category      string  `gorm:"size:100;index" json:"category,omitempty"`            // Kelompok bahan (sayur, lauk, bumbu, dll)
	DefaultUnit   string  `gorm:"not null;size:30;default:'gram'" json:"default_unit"` // Satuan default berat/isi
	DefaultWeight float64 `gorm:"type:decimal(12,2);default:0" json:"default_weight"`  // Berat/isi default per 1 kemasan
	IsActive      bool    `gorm:"default:true" json:"is_active"`
	Notes         string  `gorm:"type:text" json:"notes,omitempty"`
}

func (NutritionIngredient) TableName() string {
	return "nutrition_ingredients"
}

// NutritionMenuIngredient stores ingredient composition for one menu portion.
type NutritionMenuIngredient struct {
	ID        uint           `gorm:"primarykey" json:"id"`
	CreatedAt time.Time      `json:"created_at"`
	UpdatedAt time.Time      `json:"updated_at"`
	DeletedAt gorm.DeletedAt `gorm:"index" json:"-"`

	MenuID uint `gorm:"not null;index:idx_menu_ingredient" json:"menu_id"`

	IngredientID uint                 `gorm:"not null;index:idx_menu_ingredient" json:"ingredient_id"`
	Ingredient   *NutritionIngredient `gorm:"foreignKey:IngredientID" json:"ingredient,omitempty"`

	WeightPerPortion float64 `gorm:"type:decimal(10,2);default:0" json:"weight_per_portion"` // Berat per 1 porsi
	Unit             string  `gorm:"not null;size:30;default:'gram'" json:"unit"`            // Satuan berat/jumlah
	Notes            string  `gorm:"type:text" json:"notes,omitempty"`
}

func (NutritionMenuIngredient) TableName() string {
	return "nutrition_menu_ingredients"
}

// NutritionIngredientInvoice represents ingredient purchase invoice input in nutrition module.
// Tidak mempengaruhi stok, hanya pencatatan faktur bahan.
type NutritionIngredientInvoice struct {
	ID        uint           `gorm:"primarykey" json:"id"`
	CreatedAt time.Time      `json:"created_at"`
	UpdatedAt time.Time      `json:"updated_at"`
	DeletedAt gorm.DeletedAt `gorm:"index" json:"-"`

	Code          string                           `gorm:"not null;uniqueIndex;size:20" json:"code"`             // Kode internal faktur (NFIyyyyMMddxxx)
	InvoiceNumber string                           `gorm:"not null;size:100;index" json:"invoice_number"`        // Nomor faktur vendor
	InvoiceDate   time.Time                        `gorm:"not null;index" json:"invoice_date"`                   // Tanggal faktur
	SupplierName  string                           `gorm:"size:200;index" json:"supplier_name,omitempty"`        // Nama supplier
	ReceivedByID  *uint                            `gorm:"index" json:"received_by_id,omitempty"`                // Petugas input/terima
	ReceivedBy    *Employee                        `gorm:"foreignKey:ReceivedByID" json:"received_by,omitempty"` // Relasi petugas
	TotalAmount   float64                          `gorm:"type:decimal(15,2);default:0" json:"total_amount"`     // Total nominal faktur (computed)
	Notes         string                           `gorm:"type:text" json:"notes,omitempty"`
	Items         []NutritionIngredientInvoiceItem `gorm:"foreignKey:InvoiceID;constraint:OnUpdate:CASCADE,OnDelete:CASCADE" json:"items,omitempty"`
}

func (NutritionIngredientInvoice) TableName() string {
	return "nutrition_ingredient_invoices"
}

// NutritionIngredientInvoiceItem stores ingredient lines in one nutrition invoice.
type NutritionIngredientInvoiceItem struct {
	ID        uint           `gorm:"primarykey" json:"id"`
	CreatedAt time.Time      `json:"created_at"`
	UpdatedAt time.Time      `json:"updated_at"`
	DeletedAt gorm.DeletedAt `gorm:"index" json:"-"`

	InvoiceID uint                        `gorm:"not null;index:idx_nutrition_invoice_ingredient" json:"invoice_id"`
	Invoice   *NutritionIngredientInvoice `gorm:"foreignKey:InvoiceID" json:"-"`

	IngredientID uint                 `gorm:"not null;index:idx_nutrition_invoice_ingredient" json:"ingredient_id"`
	Ingredient   *NutritionIngredient `gorm:"foreignKey:IngredientID" json:"ingredient,omitempty"`

	Quantity    float64 `gorm:"type:decimal(12,2);default:0" json:"quantity"`       // Jumlah kemasan
	Unit        string  `gorm:"not null;size:30;default:'kemasan'" json:"unit"`     // Satuan kemasan (kemasan/pcs/bungkus)
	UnitWeight  float64 `gorm:"type:decimal(12,2);default:0" json:"unit_weight"`    // Berat/isi per kemasan (snapshot master)
	WeightUnit  string  `gorm:"not null;size:30;default:'gram'" json:"weight_unit"` // Satuan berat/isi
	TotalWeight float64 `gorm:"type:decimal(12,2);default:0" json:"total_weight"`   // quantity * unit_weight
	UnitPrice   float64 `gorm:"type:decimal(15,2);default:0" json:"unit_price"`     // Harga per kemasan
	LineTotal   float64 `gorm:"type:decimal(15,2);default:0" json:"line_total"`     // quantity * unit_price
	Notes       string  `gorm:"type:text" json:"notes,omitempty"`
}

func (NutritionIngredientInvoiceItem) TableName() string {
	return "nutrition_ingredient_invoice_items"
}

File: frontend/src/lib/api/nutrition.ts
import { api } from './client';

// ==========================================
// NUTRITION API (Gizi)
// ==========================================

// Types
export interface NutritionMenu {
  id: number;
  created_at: string;
  updated_at: string;
  code: string;
  name: string;
  description?: string;
  category: string;
  diet_types?: string; // JSON array string
  calories: number;
  protein: number;
  fat: number;
  carbohydrate: number;
  fiber: number;
  sodium: number;
  serving_size?: string;
  unit_price: number;
  is_active: boolean;
  notes?: string;
  ingredients?: NutritionMenuIngredient[];
}

export interface NutritionIngredient {
  id: number;
  created_at: string;
  updated_at: string;
  code: string;
  name: string;
  category?: string;
  default_unit: string;
  default_weight: number;
  is_active: boolean;
  notes?: string;
}

export interface NutritionMenuIngredient {
  id?: number;
  menu_id?: number;
  ingredient_id: number;
  ingredient?: NutritionIngredient;
  weight_per_portion: number;
  unit: string;
  notes?: string;
}

export interface NutritionIngredientInvoiceItem {
  id?: number;
  invoice_id?: number;
  ingredient_id: number;
  ingredient?: NutritionIngredient;
  quantity: number;
  unit: string; // satuan kemasan
  unit_weight: number; // berat/isi per kemasan
  weight_unit: string; // satuan berat/isi
  total_weight: number; // quantity * unit_weight
  unit_price: number;
  line_total: number;
  notes?: string;
}

export interface NutritionIngredientInvoice {
  id: number;
  created_at: string;
  updated_at: string;
  code: string;
  invoice_number: string;
  invoice_date: string;
  supplier_name?: string;
  received_by_id?: number;
  received_by?: any;
  total_amount: number;
  notes?: string;
  items?: NutritionIngredientInvoiceItem[];
}

export interface NutritionIngredientInvoiceInput {
  invoice_number: string;
  invoice_date: string; // YYYY-MM-DD
  supplier_name?: string;
  notes?: string;
  items: {
    ingredient_id: number;
    quantity: number;
    unit: string;
    unit_price: number;
    notes?: string;
  }[];
}

export interface NutritionPackageItem {
  id: number;
  package_id: number;
  menu_id: number;
  menu?: NutritionMenu;
  quantity: number;
  notes?: string;
}

export interface NutritionPackage {
  id: number;
  created_at: string;
  updated_at: string;
  code: string;
  name: string;
  description?: string;
  diet_type: string;
  meal_time: string;
  total_calories: number;
  total_protein: number;
  total_fat: number;
  total_carbohydrate: number;
  price: number;
  is_active: boolean;
  notes?: string;
  items?: NutritionPackageItem[];
}

export interface NutritionPackageInput {
  code?: string;
  name: string;
  description?: string;
  diet_type: string;
  meal_time: string;
  price: number;
  is_active: boolean;
  notes?: string;
  items: {
    menu_id: number;
    quantity: number;
    notes?: string;
  }[];
}

// Labels
export const nutritionCategoryLabels: Record<string, string> = {
  makanan_pokok: 'Makanan Pokok',
  lauk: 'Lauk Pauk',
  sayur: 'Sayuran',
  buah: 'Buah-buahan',
  snack: 'Makanan Ringan',
  minuman: 'Minuman',
  suplemen: 'Suplemen Gizi',
  lainnya: 'Lainnya',
};

export const nutritionDietTypeLabels: Record<string, string> = {
  biasa: 'Diet Biasa',
  lunak: 'Makanan Lunak',
  saring: 'Makanan Saring',
  cair: 'Makanan Cair',
  dm: 'Diet Diabetes Mellitus',
  rendah_garam: 'Diet Rendah Garam',
  rendah_lemak: 'Diet Rendah Lemak',
  tinggi_kalori: 'Diet Tinggi Kalori Tinggi Protein',
  rendah_protein: 'Diet Rendah Protein',
  rendah_purin: 'Diet Rendah Purin',
  lambung: 'Diet Lambung',
  jantung: 'Diet Jantung',
  ginjal: 'Diet Ginjal',
  hati: 'Diet Hati',
  lainnya: 'Diet Lainnya',
};

export const nutritionMealTimeLabels: Record<string, string> = {
  pagi: 'Makan Pagi',
  snack_pagi: 'Snack Pagi',
  siang: 'Makan Siang',
  snack_sore: 'Snack Sore',
  sore: 'Makan Sore',
  snack_malam: 'Snack Malam',
};

export const nutritionCategoryColors: Record<string, string> = {
  makanan_pokok: 'bg-amber-100 text-amber-800 dark:bg-amber-900 dark:text-amber-200',
  lauk: 'bg-red-100 text-red-800 dark:bg-red-900 dark:text-red-200',
  sayur: 'bg-green-100 text-green-800 dark:bg-green-900 dark:text-green-200',
  buah: 'bg-orange-100 text-orange-800 dark:bg-orange-900 dark:text-orange-200',
  snack: 'bg-purple-100 text-purple-800 dark:bg-purple-900 dark:text-purple-200',
  minuman: 'bg-blue-100 text-blue-800 dark:bg-blue-900 dark:text-blue-200',
  suplemen: 'bg-pink-100 text-pink-800 dark:bg-pink-900 dark:text-pink-200',
  lainnya: 'bg-gray-100 text-gray-800 dark:bg-gray-900 dark:text-gray-200',
};

export const nutritionIngredientUnitLabels: Record<string, string> = {
  gram: 'Gram',
  ml: 'Mililiter',
  pcs: 'Pcs',
  buah: 'Buah',
  sendok_makan: 'Sendok Makan',
  sendok_teh: 'Sendok Teh',
};

// API

export const nutritionMenuApi = {
  getAll: (params?: { page?: number; limit?: number; search?: string; category?: string; diet_type?: string; is_active?: string }) =>
    api.get('/nutrition/menus', { params }),

  getById: (id: number) =>
    api.get(`/nutrition/menus/${id}`),

  create: (data: Partial<NutritionMenu>) =>
    api.post('/nutrition/menus', data),

  update: (id: number, data: Partial<NutritionMenu>) =>
    api.put(`/nutrition/menus/${id}`, data),

  delete: (id: number) =>
    api.delete(`/nutrition/menus/${id}`),

  getCategories: () =>
    api.get('/nutrition/categories'),

  getDietTypes: () =>
    api.get('/nutrition/diet-types'),

  createDietType: (data: { name: string; code?: string; description?: string }) =>
    api.post('/nutrition/diet-types', data),

  getMealTimes: () =>
    api.get('/nutrition/meal-times'),

  getIngredientUnits: () =>
    api.get('/nutrition/ingredient-units'),
};

export const nutritionIngredientApi = {
  getAll: (params?: { page?: number; limit?: number; search?: string; category?: string; is_active?: string }) =>
    api.get('/nutrition/ingredients', { params }),

  getById: (id: number) =>
    api.get(`/nutrition/ingredients/${id}`),

  create: (data: Partial<NutritionIngredient>) =>
    api.post('/nutrition/ingredients', data),

  update: (id: number, data: Partial<NutritionIngredient>) =>
    api.put(`/nutrition/ingredients/${id}`, data),

  delete: (id: number) =>
    api.delete(`/nutrition/ingredients/${id}`),
};

export const nutritionIngredientInvoiceApi = {
  getAll: (params?: { page?: number; limit?: number; search?: string; start_date?: string; end_date?: string }) =>
    api.get('/nutrition/invoices', { params }),

  getById: (id: number) =>
    api.get(`/nutrition/invoices/${id}`),

  create: (data: NutritionIngredientInvoiceInput) =>
    api.post('/nutrition/invoices', data),

  update: (id: number, data: NutritionIngredientInvoiceInput) =>
    api.put(`/nutrition/invoices/${id}`, data),

  delete: (id: number) =>
    api.delete(`/nutrition/invoices/${id}`),
};

export const nutritionPackageApi = {
  getAll: (params?: { page?: number; limit?: number; search?: string; diet_type?: string; meal_time?: string; is_active?: string }) =>
    api.get('/nutrition/meal-packages', { params }),

  getById: (id: number) =>
    api.get(`/nutrition/meal-packages/${id}`),

  create: (data: NutritionPackageInput) =>
    api.post('/nutrition/meal-packages', data),

  update: (id: number, data: NutritionPackageInput) =>
    api.put(`/nutrition/meal-packages/${id}`, data),

  delete: (id: number) =>
    api.delete(`/nutrition/meal-packages/${id}`),
};

// ==========================================
// Nutrition Order Types & API
// ==========================================

export interface NutritionOrderItem {
  id: number;
  order_id: number;
  menu_id: number;
  menu?: NutritionMenu;
  quantity: number;
  notes?: string;
}

export interface NutritionOrder {
  id: number;
  created_at: string;
  updated_at: string;
  visit_id: number;
  visit?: any;
  patient_id: number;
  patient?: any;
  order_date: string;
  meal_time: string;
  diet_type: string;
  status: string;
  package_id?: number;
  package?: NutritionPackage;
  room_name?: string;
  bed_name?: string;
  ordered_by_id?: number;
  ordered_by?: any;
  prepared_at?: string;
  delivered_at?: string;
  delivered_by_id?: number;
  allergy_notes?: string;
  special_notes?: string;
  items?: NutritionOrderItem[];
}

export interface CreateNutritionOrderInput {
  visit_id: number;
  meal_time: string;
  diet_type: string;
  order_date: string;
  package_id?: number;
  allergy_notes?: string;
  special_notes?: string;
  items?: {
    menu_id: number;
    quantity: number;
    notes?: string;
  }[];
}

export interface NutritionIngredientUsageRow {
  ingredient_id: number;
  ingredient_code: string;
  ingredient_name: string;
  ingredient_category?: string;
  unit: string;
  total_usage: number;
}

export interface NutritionIngredientUsageSummary {
  rows: number;
  statuses: string[];
  start_date: string;
  end_date: string;
  meal_time?: string;
  diet_type?: string;
  room_name?: string;
  generated_at: string;
}

export const nutritionOrderStatusLabels: Record<string, string> = {
  draft: 'Draft',
  confirmed: 'Dikonfirmasi',
  preparing: 'Sedang Disiapkan',
  delivered: 'Sudah Diantar',
  cancelled: 'Dibatalkan',
};

export const nutritionOrderStatusColors: Record<string, string> = {
  draft: 'bg-gray-100 text-gray-800 dark:bg-gray-900 dark:text-gray-200',
  confirmed: 'bg-blue-100 text-blue-800 dark:bg-blue-900 dark:text-blue-200',
  preparing: 'bg-yellow-100 text-yellow-800 dark:bg-yellow-900 dark:text-yellow-200',
  delivered: 'bg-green-100 text-green-800 dark:bg-green-900 dark:text-green-200',
  cancelled: 'bg-red-100 text-red-800 dark:bg-red-900 dark:text-red-200',
};

export const nutritionOrderApi = {
  getAll: (params?: { visit_id?: number; patient_id?: number; date?: string; status?: string; meal_time?: string; room_name?: string; page?: number; limit?: number }) =>
    api.get('/nutrition/orders', { params }),

  getById: (id: number) =>
    api.get(`/nutrition/orders/${id}`),

  create: (data: CreateNutritionOrderInput) =>
    api.post('/nutrition/orders', data),

  updateStatus: (id: number, status: string) =>
    api.put(`/nutrition/orders/${id}/status`, { status }),

  delete: (id: number) =>
    api.delete(`/nutrition/orders/${id}`),

  getKitchenDashboard: (date?: string) =>
    api.get('/nutrition/kitchen', { params: { date } }),
};

export const nutritionReportApi = {
  getIngredientUsage: (params?: { start_date?: string; end_date?: string; status?: string; meal_time?: string; diet_type?: string; room_name?: string }) =>
    api.get('/nutrition/reports/ingredient-usage', { params }),
};
